File: api/internal/models/trade.go
package models

import "time"

type Trade struct {
    ID        uint      `gorm:"primaryKey"`
    Symbol    string
    Side      string
    Price     float64   // per unit price
    Quantity  float64
    Fee       float64   // in fee asset
    FeeAsset  string    // e.g. "USDT", "BTC"
    Exchange  string
    Timestamp time.Time
}

File: dashboard/src/types/trade.ts
export type Trade = {
  id: number
  botID: string
  symbol: string
  side: string // BUY / SELL / NONE
  price: number // per unit price
  quantity: number
  fee: number // in fee asset
  feeAsset: string // e.g. "USDT", "BTC"
  exchange: string
  timestamp: Date
  createdAt: Date // GORM convention
}

export type TradeFilters = {
  limit: number
  symbol: string
  side: string
  botId: string
  dateFrom: string
  dateTo: string
}
